### PR TITLE
[EI-109] Interest function modifications

### PR DESCRIPTION
**Pull request by jspellman814**
Continuing the discussion/implementation from #22 around renaming interest-related functions.

https://github.com/pantheon-systems/pantheon-wordpress-edge-integrations/pull/22#issuecomment-1040458217

> We should consider renaming some of the wrapper functions. For example, renaming set_interest to set_vary_header (or something similar) since you can pass whatever key you want to the varyHeader. And then maybe rename get_interest as well since you can both get and set the interest using that one function (though I don't have a great suggestion for an alternative name at the moment).
> 
> Or perhaps we rename get_interest to set_interest, write a new function for getting the interest data (where that's all you can do), and rename the current set_interest to set_vary_header. 

Open to feedback, but at the moment this PR implements the latter suggestion from above.

### CODE REVIEW

**Comment by jspellman814** [comment]
This makes more sense to me after our call today. I removed the `set_vary_header` function based on the discussion, and the limited scope of get/set functions. Going to mark as draft in case we revisit in a future sprint.

File: inc/namespace.php [hunk 1]
```diff
@@ -103,3 +105,23 @@ function set_vary_headers() {
...
+
+/**
+ * Add header key and custom data.
+ *
+ * @param array $key Key for the header, or array of keys.
+ * @param array $data Data to pass to the HeaderData class.
+ *
+ * @return array The header data.
+ */
+function add_header_data( array $key = null, array $data = null ) : array {
```
Comment (jspellman814) [inline inc/namespace.php:117]: @jazzsequence Curious of your thoughts about this function name. Struggled with naming and open to any suggestions.

Comment (jazzsequence) [reply inc/namespace.php:117]: This adds support for new vary headers, yeah? Maybe `update_vary_headers`?

Comment (jspellman814) [reply inc/namespace.php]: Yeah, that sounds better IMO.

**Review by jazzsequence — APPROVED** [review]
lgtm

**Pull request merged**